src/turbomind/models/llama/LlamaBatch.cc

```diff
@@ -258,6 +275,53 @@ void LlamaBatch<T>::ProcessInferRequests(const Requests& requests)
             output_ids = Copy(input_ids, input_length, output_ids);
         }
 
+        // copy input embeddings
+        if (r->inputs[rank_].isExist("embedding_counts")) {
+            int        emb_count    = r->inputs[rank_].getVal<int>("embedding_counts");
+            const auto emb_tensor   = r->inputs[rank_].at("input_embeddings");
+            const auto begin_tensor = r->inputs[rank_].at("embedding_begins");
+            const auto end_tensor   = r->inputs[rank_].at("embedding_ends");
```

Comment:
It is overly verbose to use 3 parameters for just specifying the range of input embeddings.

Use a single `input_embedding_ranges` with shape `[batch_size, max_input_embedding_count, 2]`, store the `begin`/`end` pairs in the inner most dimension and pad to `max_input_embedding_count` with `(-1, -1)`